tests/testAutopas/AutoTunerTest.cpp

```diff
@@ -0,0 +1,76 @@
+/**
+ * @file AutoTunerTest.cpp
+ * @author F. Gratl
+ * @date 8/10/18
+ */
+
+#include "AutoTunerTest.h"
+#include <autopas/selectors/AutoTuner.h>
+
+TEST_F(AutoTunerTest, testTune) {
+  autopas::LJFunctor<Particle, FPCell> functor;
+  std::vector<autopas::ContainerOptions> containers = {autopas::ContainerOptions::verletLists,
+                                                       autopas::ContainerOptions::directSum,
+                                                       autopas::ContainerOptions::linkedCells};
+  std::vector<autopas::TraversalOptions> traversals = {autopas::TraversalOptions::sliced,
+                                                       autopas::TraversalOptions::c08};
+
+  std::array<double, 3> bBoxMin = {0, 0, 0}, bBoxMax = {10, 10, 100};
+  const double cutoff = 1;
+  const double verletSkin = 0;
+  const unsigned int verletRebuildFrequency = 1;
+  autopas::AutoTuner<Particle, FPCell> autoTuner(bBoxMin, bBoxMax, cutoff, verletSkin, verletRebuildFrequency,
+                                                 containers, traversals, 100);
+
+  std::shared_ptr<autopas::ParticleContainer<Particle, FPCell>> fastestContainer;
+  AutoPasLogger->set_level(spdlog::level::debug);
+  bool stillTuning = true;
+  int i = 0;
+  for (; stillTuning; ++i) {
+    AutoPasLogger->debug("AutoTunerTest: Iteration {}", i);
+    stillTuning = autoTuner.iteratePairwise(&functor, autopas::DataLayoutOption::aos);
+
+    auto container = autoTuner.getContainer();
+
+    // tuning phases:
+    // 0 -> test verlet
+    // 1 -> test directSum
+    // 2 -> test linked with sliced
+    // 3 -> test linked with c08
+    // 4 -> choose best lc traversal -> traversal tuning finished
```

Comment:
why two extra phases? shouldn't it be enough to go until 3?
4 and 5 seem redundant

Reply:
4 -> finished tuning LC traversals and finding fastest
5 -> finished tuning Containers and finding fastest
Both need a dedicated iteration since the results are not in when selecting the last.
One could theoretically get rid of those iterations but this might require more rewriting than this kind of AutoTuning is worth.